org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/utils/XMLPositionUtility.java

```diff
@@ -997,6 +997,21 @@ public static DocumentLink createDocumentLink(DOMRange target, String location)
 		return new DocumentLink(new Range(start, end), location);
 	}
 
+	/**
+	 * Create a document link without adjusting the range to account for quotes.
+	 * 
+	 * @param target The range in the document that should be the link
+	 * @param location URI where the link should point
+	 * @return A DocumentLink over the range <code>target</code> that points to <code>location</code>.
+	 * @throws BadLocationException if <code>target</code> is out of the bounds of the document
+	 */
+	public static DocumentLink createDocumentLinkNoAdjust(DOMRange target, String location) throws BadLocationException {

```

Comment:
I will update the existing createDocumentLink to `createDocumentLink(DOMRange target, String location, boolean adjust)` and createDocumentLink((DOMRange target, String location) will call `createDocumentLink(target, location, true)` to share the same code.